ui/address/AddressTokens.tsx

```diff
@@ -58,111 +61,94 @@ const AddressTokens = () => {
-    updateTokensData('ERC-20', payload);
-  }, [ updateTokensData ]);
-
-  const handleTokenBalancesErc721Message: SocketMessage.AddressTokenBalancesErc721['handler'] = React.useCallback((payload) => {
-    updateTokensData('ERC-721', payload);
-  }, [ updateTokensData ]);
-
-  const handleTokenBalancesErc1155Message: SocketMessage.AddressTokenBalancesErc1155['handler'] = React.useCallback((payload) => {
-    updateTokensData('ERC-1155', payload);
-  }, [ updateTokensData ]);
-
-  const channel = useSocketChannel({
-    topic: `addresses:${ hash.toLowerCase() }`,
-    isDisabled: erc20Query.isPlaceholderData || erc721Query.isPlaceholderData || erc1155Query.isPlaceholderData,
-  });
-
-  useSocketMessage({
-    channel,
-    event: 'updated_token_balances_erc_20',
-    handler: handleTokenBalancesErc20Message,
-  });
-  useSocketMessage({
-    channel,
-    event: 'updated_token_balances_erc_721',
-    handler: handleTokenBalancesErc721Message,
-  });
-  useSocketMessage({
-    channel,
-    event: 'updated_token_balances_erc_1155',
-    handler: handleTokenBalancesErc1155Message,
-  });
+  const handleNFTsDisplayTypeChange = React.useCallback((val: TNftDisplayType) => {
+    cookies.set(cookies.NAMES.ADDRESS_NFT_DISPLAY_TYPE, val);
+    setNftDisplayType(val);
+  }, []);
+
+  const handleTokenTypesChange = React.useCallback((value: Array<NFTTokenType>) => {
+    nftsQuery.onFilterChange({ type: value });
+    collectionsQuery.onFilterChange({ type: value });
+    setTokenTypes(value);
+  }, [ nftsQuery, collectionsQuery ]);
+
+  const nftTypeFilter = (
+    <PopoverFilter isActive={ tokenTypes && tokenTypes.length > 0 } contentProps={{ w: '200px' }} appliedFiltersNum={ tokenTypes?.length }>
+      <TokenTypeFilter<NFTTokenType> nftOnly onChange={ handleTokenTypesChange } defaultValue={ tokenTypes }/>
+    </PopoverFilter>
+  );
 
   const tabs = [
-    { id: tokenTabsByType['ERC-20'], title: 'ERC-20', component: <ERC20Tokens tokensQuery={ erc20Query }/> },
-    { id: tokenTabsByType['ERC-721'], title: 'ERC-721', component: <ERC721Tokens tokensQuery={ erc721Query }/> },
-    { id: tokenTabsByType['ERC-1155'], title: 'ERC-1155', component: <ERC1155Tokens tokensQuery={ erc1155Query }/> },
+    { id: 'tokens_erc20', title: 'ERC-20', component: <ERC20Tokens tokensQuery={ erc20Query }/> },
+    {
+      id: 'tokens_nfts',
+      title: 'NFTs',
+      component: nftDisplayType === 'list' ?
+        <AddressNFTs tokensQuery={ nftsQuery }/> :
+        <AddressCollections collectionsQuery={ collectionsQuery } address={ hash }/>,
+    },
   ];
 
+  const nftDisplayTypeRadio = (
+    <RadioButtonGroup<TNftDisplayType>
+      onChange={ handleNFTsDisplayTypeChange }
+      defaultValue={ nftDisplayType }
+      name="type"
+      options={ [
+        { title: 'By collection', value: 'collection', icon: collectionIcon, onlyIcon: isMobile },
```

Comment:
Why don't we have `onlyIcon` props for the component itself instead of for each option? Could one of the options be with just an icon and another with text and an icon?

Reply:
maybe it can, but I chose this way because of typescript

Reply:
I don't really get why Typescript forced you to have separate flags for each option instead of just one prop for the whole component.